### PR TITLE
Rholang: Remove Channel and Quote from the proto

### PR DESCRIPTION
**Pull request by guardbotmk3**
They aren't necessary, and they complicate working with the protos.

Fixed everywhere they are used, and all tests pass after the removal. After this
commit, you will need to run sbt rholangProtoBuild/clean casper/clean at a
minimum. This a breaking change for the term encoding, which is why I'm trying
to get it in before the release.

### CODE REVIEW

**Comment by goral09** [comment]
I like that `Quote` was an explicit way of saying - "here, this is a name now". It felt that we are closer tho rho calculus.

**Comment by goral09** [comment]
Hmm, I don't see the removal of `Quote` from the `RhoTypes.proto` @guardbotmk3 .

File: models/src/main/protobuf/RhoTypes.proto [hunk 1]
```diff
@@ -36,20 +36,13 @@ message ParWithRandom {
     bytes randomState = 2 [(scalapb.field).type = "coop.rchain.crypto.hash.Blake2b512Random"];
 }
 
-message Channel {
-    oneof channel_instance {
-        Par quote = 1;
-        Var chanVar = 2;
-    }
-}
-
```
Comment (guardbotmk3) [inline models/src/main/protobuf/RhoTypes.proto]: @goral09 It's removed right here.

**Comment by guardbotmk3** [comment]
> I like that Quote was an explicit way of saying - "here, this is a name now". It felt that we are closer tho rho calculus.

I like it for user input
But once we've validated the user input, it's not necessary anymore. When writing scala, we needed to pull things apart twice, rather than once. Look at how much it shortens the registry.

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala [hunk 2]
```diff
@@ -70,7 +69,7 @@ class CollectMatcherSpec extends FlatSpec with Matchers {
     result.par should be(
       inputs.par.prepend(
         EList(
-          List[Par](EVar(BoundVar(1)), EEvalBody(ChanVar(BoundVar(0))), GInt(7)),
+          List[Par](EVar(BoundVar(1)), EVar(BoundVar(0)), GInt(7)),
```
Comment (goral09) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala]: I am really glad we are getting rid of this. I didn't like that `BoundVar`s could be existing both in a `Channel` and as a `Var`.

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala [hunk 3]
```diff
@@ -192,8 +191,8 @@ class ProcPrinterSpec extends FlatSpec with Matchers {
       )
     val target =
       """new x0 in {
-        |  for( x1 <- x0 ) {
-        |    *x1
+        |  for( @{x1} <- @{x0} ) {
```
Comment (goral09) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala]: So it seems that previously it was printing them wrong?

**Pull request merged**